draft-ietf-ppm-dap.md

```diff
@@ -654,19 +654,29 @@ configuration of each aggregator. See {{compliance}} for information on HPKE
 algorithm choices.
 
 Clients retrieve the HPKE configuration from each aggregator by
-sending an HTTP GET request to `[aggregator]/hpke_config?task_id=[task-id]`, where
+sending an HTTP GET request to `[aggregator]/hpke_config`, where
 `[aggregator]` is the aggregator's endpoint URL, obtained from the task
-parameters, and `[task-id]` is the task ID obtained from the task parameters,
+parameters. A deployment MAY choose to use different HPKE configurations
+for different tasks, so clients can optionally specify a query parameter `task_id`
+when sending an HTTP GET request to `[aggregator]/hpke_config?task_id=[task-id]`,
+where `[task-id]` is the task ID obtained from the task parameters,
 encoded in Base 64 with URL and filename safe alphabet with no padding, as
 specified in sections 5 and 3.2 of {{!RFC4648}}. If the aggregator does not
 recognize the task ID, then it responds with HTTP status code 404 Not Found and
-an error of type `unrecognizedTask`. The aggregator responds to well-formed
-requests with HTTP status code 200 OK and an `HpkeConfig` value:
+an error of type `unrecognizedTask`. If a deployment enforces different HPKE
+configurations for different tasks, and a client doesn't specify `task_id` in the
+HTTP GET request, then the aggregator responds with HTTP status code 400 Bad Request
+and an error of type `unrecognizedMessage`. The aggregator responds to well-formed
```

Comment:
`unrecognizedMessage` and HTTP 400 seem inappropriate. The problem isn't an incorrectly encoded message, it's that this particular aggregator chooses not to implement this functionality. If a client does `GET /hpke_config` and the server doesn't support it, it should get some indication that it should now try `GET /hpke_config?task_id=[task_id]`.

[HTTP 422](https://developer.mozilla.org/en-US/docs/Web/HTTP/Status/422) seems pretty appropriate for this case ("the server understands the content type of the request entity, and the syntax of the request entity is correct, but it was unable to process the contained instructions"). I think we should use that status code and define a new problem document type (in section 3.1 errors) for this specific case.

Reply:
+1 to new abort type (e.g., "missingTaskId"), but @tgeoghegan, we currently we return the same status code for all abort cases (400). It might be a good idea to let the status code differ for different abort types, but we should leave that out of this change I think.

Reply:
That's fair enough. The problem document type will make this explicit enough, and [RFC 9205](https://httpwg.org/specs/rfc9205.html#using-http-status-codes) recommends using the most general possible HTTP status anyway (in this case, 400).

Reply:
Sounds good, I added a new `missingTaskId` problem type under {{Errors}}.